tpm/tpm.go

```diff
@@ -940,6 +940,39 @@ func NVReadValue(rw io.ReadWriter, index, offset, len uint32, ownAuth []byte) ([
 	return data, nil
 }
 
+// NVWriteValue for writing to the NVRAM. Needs a index for a defined space in NVRAM.
+// See TPM-Main-Part-3-Commands_v1.2_rev116_01032011, P216
+func NVWriteValue(rw io.ReadWriter, index, offset uint32, data []byte, ownAuth []byte) error {
+	if ownAuth == nil {
+		_, _, _, err := nvWriteValue(rw, index, offset, uint32(len(data)), data, nil)
+		if err != nil {
+			return fmt.Errorf("failed to read from NVRAM: %v", err)
+		}
+		return err
```

Comment:
Should be okay to `return nil` here, as the error case is already handled above.

Reply:
One idea:

```go
if _, _, _, err := nvWriteValue(rw, index, offset, uint32(len(data)), data, nil); err != nil {
	return fmt.Errorf("failed to read from NVRAM: %v", err)
}
return nil
```